src/Couchbase.Lite.Tests.Shared/ScopeCollectionTest.cs

```diff
@@ -89,5 +89,277 @@ public void TestGetDefaultScopeAfterDeleteDefaultCollection()
+            for (char letter = '!'; letter <= '/'; letter++) {
+                if (letter == '%')
+                    return;
+                Action badAction = (() => Db.CreateCollection(str + letter));
+                badAction.Should().Throw<CouchbaseLiteException>($"Invalid collection name '{str + letter}' in scope '_default'.");
+            }
+
+            for (char letter = '<'; letter <= '@'; letter++) {
+                Action badAction = (() => Db.CreateCollection(str + letter));
+                badAction.Should().Throw<CouchbaseLiteException>($"Invalid collection name '{str + letter}' in scope '_default'.");
+            }
+
+            for (char letter = '['; letter <= '`'; letter++) {
+                Action badAction = (() => Db.CreateCollection(str + letter));
+                badAction.Should().Throw<CouchbaseLiteException>($"Invalid collection name '{str + letter}' in scope '_default'.");
+            }
+
+            for (char letter = '{'; letter <= '~'; letter++) {
+                Action badAction = (() => Db.CreateCollection(str + letter));
+                badAction.Should().Throw<CouchbaseLiteException>($"Invalid collection name '{str + letter}' in scope '_default'.");
+            }
+        }
+
+        [Fact]
+        public void TestCollectionNameLength()
+        {
+            /* Collection name should contain valid char with length from 1 to 251 */
+            var collName = "";
+            for (int i = 0; i < 251; i++) {
+                collName += 'c';
+                collName.Length.Should().Be(i + 1);
+                Db.CreateCollection(collName).Should().NotBeNull($"Valid collection '{collName}' length {collName.Length}.");
+            }
+
+            var str = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_%-cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc";
+            str.Length.Should().Be(251);
+            Db.CreateCollection(str).Should().NotBeNull($"because the collection name can be length at {str.Length}");
+
+            str += "e";
+            str.Length.Should().Be(252);
+            Action badAction = (() => Db.CreateCollection(str));
+            badAction.Should().Throw<CouchbaseLiteException>($"Invalid collection name '{str}' in scope because the collection name length {str.Length} is over naming length limit.");
+        }
+
+        [Fact] // I think this test case is a dup of TestCollectionNameWithValidChars?
+        public void TestCollectionNameCaseSensitive()
+        {
+            Db.CreateCollection("COLLECTION1").Should().NotBeNull();
+            // TODO : wait for CBL-3195 fix
+            //Db.CreateCollection("collection1").Should().NotBeNull("Should be able to be created because collection name is case sensitive.");
```

Comment:
This will already return non-null, it will just be the same as COLLECTION1.  The test should ensure they are different from each other (e.g. independent documents)